From 9d1d02716aef83c20182ed066b082d1d2680b9d3 Mon Sep 17 00:00:00 2001
From: Hiren Vadukul <hirenv@ext.mechasystems.com>
Date: Thu, 11 Sep 2025 17:16:27 +0530
Subject: [PATCH] gpio-keys: Add keypad slide switch event support

Add keypad slide event to gpio-keys node for Comet Extension Keypad Slide

Signed-off-by: Hiren Vadukul <hirenv@ext.mechasystems.com>

diff --git a/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts b/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
index 7e3c0dfb8cca..f03108bfc801 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-mecha-comet-m-gen6.dts
@@ -142,6 +142,17 @@ usb3_data_ss: endpoint {
 			};
 		};
 	};
+
+	Comet-Extension {
+		compatible = "gpio-keys";
+
+		keypad-slide {
+			label = "Comet Extension Keypad Slide";
+			linux,input-type = <EV_SW>;
+			linux,code = <SW_KEYPAD_SLIDE>;
+			gpios = <&gpio4 23 GPIO_ACTIVE_HIGH>;
+		};
+	};
 };
 
 &usb3_0 {
-- 
2.25.1

